From 7d9e7f13fc7147b1d7cd4718c2df4328b578a23a Mon Sep 17 00:00:00 2001
From: Johannes Kauffmann <johanneskauffmann@hotmail.com>
Date: Sun, 31 Oct 2021 11:13:25 +0100
Subject: [PATCH] lua: discard basedir and DIR_SEP on UWP

On UWP, the lua folder is deployed right next to the executable in the
app installation folder, but the path returned by config_GetUserDir
(which is basedir) points to the local packages folder, located in the
current user AppData folder. Therefore, discard basedir and the
directory separator, to allow VLC to find the lua folder.
---
 modules/lua/vlc.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/modules/lua/vlc.c b/modules/lua/vlc.c
index f20f931f189..ab0e5a4f334 100644
--- a/modules/lua/vlc.c
+++ b/modules/lua/vlc.c
@@ -183,8 +183,12 @@ static char **vlclua_dir_list_append( char **restrict list, char *basedir,
     if (unlikely(basedir == NULL))
         return list;
 
+#ifndef VLC_WINSTORE_APP
     if (likely(asprintf(list, "%s"DIR_SEP"lua"DIR_SEP"%s",
                         basedir, luadirname) != -1))
+#else
+    if (likely(asprintf(list, "lua"DIR_SEP"%s", luadirname) != -1))
+#endif
         list++;
 
     free(basedir);
-- 
GitLab

